overlord/ifacestate/helpers.go

```diff
@@ -526,3 +528,54 @@ func snapsWithSecurityProfiles(st *state.State) ([]*snap.Info, error) {
 
 	return infos, nil
 }
+
+// SnapInterfaceState describes state of interfaces of a single snap.
```

Comment:
We have quite a lot of state for the interfaces of a snap already, and that's not related to any of it. I don't quite understand the details of the bug yet, so I may certainly be off, but given that completely parallel implementation, it tastes like an isolated workaround for a very specific bug that should be properly considered in the existing logic instead of besides it.

Let's please catch up tomorrow on this.